nservicebus/outbox/index.md

```diff
@@ -22,11 +22,9 @@ Most message queues, and some data stores, do not support distributed transactio
 
 The NServiceBus **outbox** feature ensures consistency between business data and messages. It simulates an atomic transaction, distributed across both the data store used for business data and the message queue used for messaging.
 
-Note: Messages dispatched to the transport as part of the outbox dispatch stage will not be batched and each message is sent in isolation.
```

Comment:
I think the more accurate wording would be something like

> Note: Messages dispatched to the transport when the outbox is enabled will not be batched and each message is sent in isolation.

Reply:
and I agree this should be moved but I would keep the note

Reply:
> Messages dispatched to the transport when the outbox is enabled will not be batched and each message is sent in isolation.

how does the outbox behavior differ from regular dispatching? It uses the same pipeline and everything, so I'm not sure what information we want to give to the user here?

Reply:
It does differ because of this behavior https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Reliability/Outbox/ForceBatchDispatchToBeIsolatedBehavior.cs#L19 and this is what the note is about according to my understanding

Reply:
I raised https://github.com/Particular/NServiceBus.AmazonSQS/issues/2248 and https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/issues/857

Reply:
> As far as I recall the Isolated has historically been interpreted differently than how you described it here.

but the scenario here would be:

1. Incoming message
2. Invoke handlers
3. handlers send message
4. message gets added to outgoing messages batch
5. Outbox persists the outgoing message batch to the DB
6. Outbox transaction committed
7. Outgoing messages (batch) is set to isolated to avoid the incoming message transaction
8. Outgoing messages are dispatched
9. Outbox records are set as dispatched

from a user perspective, the observable behavior is the same as with regular batching, as even the isolated setting does not send the messages any sooner than before (it's not the immediate dispatch path used by message handlers). So the docs message:

> Messages dispatched to the transport as part of the outbox dispatch stage will not be batched and each message is sent in isolation.

is technically true from a transport perspective but that's only due to the transport's implementation of the isolated dispatch behavior. The only thing that might differ is that a dispatch failure might roll back all outgoing messages in a batch instead of aborting in the middle but there is nothing the user can take into account in this regard as the receiver side needs to be able to deal with duplicates anyway.

Is my understanding correct @danielmarbach ?

Reply:
> from a user perspective, the observable behavior is the same as with regular batching, as even the isolated setting does not send the messages any sooner than before (it's not the immediate dispatch path used by message handlers)

Correct that is my understanding too see 

> Rereading that PR and your arguments here I do agree now that it is probably better to remove the note since there is no impact for the user

The most important distinction is only like you said that the sends are not enlisted in the receive transaction and I think that is already mentioned somewhere in the docs right?